cmd/create/machinepool/cmd.go

```diff
@@ -17,255 +17,100 @@ limitations under the License.
+func CreateMachinepoolRunner(userOptions *mpOpts.CreateMachinepoolUserOptions) rosa.CommandRunner {
+	return func(ctx context.Context, r *rosa.Runtime, cmd *cobra.Command, argv []string) error {
+		var err error
+		options := NewCreateMachinepoolOptions()
+		clusterKey := r.GetClusterKey()
+		options.args = userOptions
+
+		cluster := r.FetchCluster()
+		if err := validateClusterState(cluster, clusterKey); err != nil {
+			return err
+		}
 
-	flags.StringSliceVar(&args.SecurityGroupIds,
-		securitygroups.MachinePoolSecurityGroupFlag,
-		nil,
-		"The additional Security Group IDs to be added to the machine pool. "+
-			"Format should be a comma-separated list.",
-	)
+		val, ok := cluster.Properties()[properties.UseLocalCredentials]
+		useLocalCredentials := ok && val == "true"
 
-	flags.StringVar(&args.NodeDrainGracePeriod,
-		"node-drain-grace-period",
-		"",
-		"You may set a grace period for how long Pod Disruption Budget-protected workloads will be "+
-			"respected when the NodePool is being replaced or upgraded.\nAfter this grace period, all remaining workloads "+
-			"will be forcibly evicted.\n"+
-			"Valid value is from 0 to 1 week (10080 minutes), and the supported units are 'minute|minutes' or "+
-			"'hour|hours'. 0 or empty value means that the NodePool can be drained without any time limitations.\n"+
-			"This flag is only supported for Hosted Control Planes.",
-	)
+		if err := validateLabels(cmd, options.args); err != nil {
+			return err
+		}
 
-	flags.StringSliceVar(
-		&args.Tags,
-		"tags",
-		nil,
-		"Apply user defined tags to all resources created by ROSA in AWS. "+
-			"Tags are comma separated, for example: 'key value, foo bar'",
-	)
+		r.AWSClient, err = aws.NewClient().
+			Region(cluster.Region().ID()).
+			Logger(r.Logger).
+			UseLocalCredentials(useLocalCredentials).
+			Build()
+		if err != nil {
+			return fmt.Errorf("Failed to create awsClient: %s", err)
+		}
+		service := NewCreateMachinePool(CreateMachinePoolSpec{
+			Service: machinepool.NewMachinePoolService(),
+		})
 
-	interactive.AddFlag(flags)
-	output.AddFlag(Cmd)
+		return service.createMachinePoolBasedOnClusterType(r, cmd, clusterKey, cluster, options.Machinepool())
+	}
 }
 
-func run(cmd *cobra.Command, _ []string) {
-	r := rosa.NewRuntime().WithOCM()
-	defer r.Cleanup()
-
-	clusterKey := r.GetClusterKey()
-
-	cluster := r.FetchCluster()
+// Validate the cluster's state is ready
+func validateClusterState(cluster *cmv1.Cluster, clusterKey string) error {
 	if cluster.State() != cmv1.ClusterStateReady {
-		r.Reporter.Errorf("Cluster '%s' is not yet ready", clusterKey)
-		os.Exit(1)
+		return fmt.Errorf("Cluster '%s' is not yet ready", clusterKey)
 	}
+	return nil
+}
 
-	val, ok := cluster.Properties()[properties.UseLocalCredentials]
-	useLocalCredentials := ok && val == "true"
-
+// Parse labels if the 'labels' flag is set
+func validateLabels(cmd *cobra.Command, args *mpOpts.CreateMachinepoolUserOptions) error {
```

Comment:
@den-rgb I think we should move this into the `Bind` method on the `CreateMachinePoolOptions`. That method would validate that the user-specified labels are in the correct format, and return an `error` if not.

We can then add a unit test for that as part of your options unit testing.

Reply:
I don't know how necessary this is considering if i was to move it to Bind all bind would be checking is the validity of the labels which would be the exact same as having this function in place